frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -124,16 +124,13 @@ export default class Table extends Component {
       section: t`Columns`,
       title: t`Cell column`,
       widget: "field",
-      getDefault: (
-        [
-          {
-            data: { cols, rows },
-          },
-        ],
-        settings,
-      ) => {
-        const col = cols.filter(isMetric)[0];
-        return col && col.name;
+      getDefault: ([{ data }], { "table.pivot_column": pivotCol }) => {
+        // We try to show numeric values in pivot cells, but if none are
+        // available, we fall back to the last column in the unpivoted table
+        const nonPivotCols = data.cols.filter(c => c.name !== pivotCol);
+        const lastCol = nonPivotCols[nonPivotCols.length - 1];
+        const [{ name } = lastCol] = nonPivotCols.filter(isMetric);
```

Comment:
This destructuring kind of weirds me out, but not a big deal. I think

```
const { name } = nonPivotCols.find(isMetric) || lastCol;
```

would be a bit clearer.

Reply:
👍 that's much better